### PR TITLE
[10.x.x] Shaders: Fixing Array defines with precisions and removing prefer_hlslcc and d3d11_9x

### PR DESCRIPTION
**Pull request by ellioman**
# Purpose of this PR
This PR fixes the following in URP Shaders:
- Removes any reference to `d3d11_9x` as support for DX9 was removed in 2018.2.0a2
- Removes any calls to `prefer_hlslcc`as that has been unused for a long time
- In 2018.3.0a2 there was support added for Texture2DArray with _float/_half precision

# Yamato
https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics/tree/universal%252Fshader-texture-array-precision

# Testing status

**Manual Tests**: What did you do?
- [x] Opened test project + Run graphic tests locally
- [ ] Built a player
- [ ] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo + Prefab overrides + Alignment in Preset
- [ ] C# and shader warnings (supress shader cache to see them)
- [ ] Checked new resources path for the reloader (in developer mode, you have a button at end of resources that check the paths)
- Other:

### CODE REVIEW

**Review by aleks01010101 — COMMENTED** [review]
Is it intentional that you excluded gles3 and glcore from many #pragma target 4.5 subshaders?
Like, if you are on GLES3/GLCore, you can't use hybrid v2?

**Review by aleks01010101 — APPROVED** [review]

File: com.unity.render-pipelines.universal/Shaders/Lit.shader [hunk 1]
```diff
@@ -84,7 +84,7 @@ Shader "Universal Render Pipeline/Lit"
             Cull[_Cull]
 
             HLSLPROGRAM
-            #pragma exclude_renderers d3d11_9x gles
+            #pragma exclude_renderers gles gles3 glcore
```
Comment (eh-unity) [inline com.unity.render-pipelines.universal/Shaders/Lit.shader:87]: Why glcore? is that correct?

**Review by eh-unity — APPROVED** [review]
Also ComplexLit.shader is now in the master. Perhaps you would need to update that too.
Otherwise it looks good to me.

File: com.unity.render-pipelines.universal/Shaders/Lit.shader (hunk 1 above)
Comment (ellioman) [reply com.unity.render-pipelines.universal/Shaders/Lit.shader:87]: Hybrid doesn't support OpenGL or GLES so we exclude those and have the second subshader only for them.

**Review by phi-lira — APPROVED** [review]

**Comment by phi-lira** [comment]
LGTM. Make sure ABV is green on URP side.

**Comment by ellioman** [comment]
The failures on Yamato (URP & HDRP) are contained to Metal on Standalone which is a known timeout issue that's being looked at. Not caused by this PR. I'm therefore merging this PR.

**Pull request merged**